tests/blackbox/stratisd_cert.py

```diff
@@ -477,6 +477,48 @@ def test_filesystem_create_permissions(self):
 
         self._test_permissions(StratisDbus.fs_create, [pool_path, fs_name], True)
 
+    def test_filesystem_udev_symlink(self):
+        """
+        Test the udev symlink creation for filesystem devices.
+        """
+        pool_name = p_n()
+        pool_path = make_test_pool(pool_name, StratisCertify.DISKS[0:1])
+
+        fs_name = fs_n()
+        filesystem_path = make_test_filesystem(pool_path, fs_name)
+
+        objects = StratisDbus.get_managed_objects()
+
+        pool_gmodata = objects[pool_path]
+        pool_uuid = pool_gmodata["org.storage.stratis2.pool"]["Uuid"]
+        filesystem_gmodata = objects[filesystem_path]
+        filesystem_uuid = filesystem_gmodata["org.storage.stratis2.filesystem"]["Uuid"]
+
+        filesystem_devnode = (
+            str(filesystem_gmodata["org.storage.stratis2.filesystem"]["Devnode"])
+            .replace("$", "_")
+            .replace("!", "_")
+        )
+
+        fs_devmapperlinkstr = (
+            "/dev/mapper/stratis-1-" + pool_uuid + "-thin-fs-" + filesystem_uuid
+        )
+
+        fsdevdest = os.path.abspath(
+            os.path.join(
+                os.path.dirname(filesystem_devnode),
+                os.readlink(str(filesystem_devnode)),
```

Comment:
You can definitely drop the use of ```str()``` here.

Reply:
Resolved.